test/rsa_sp800_56b_test.c

```diff
@@ -430,9 +438,17 @@ static int test_invalid_keypair(void)
           && TEST_ptr(e = bn_load_new(cav_e, sizeof(cav_e)))
           && TEST_ptr(n = bn_load_new(cav_n, sizeof(cav_n)))
           && TEST_ptr(d = bn_load_new(cav_d, sizeof(cav_d)))
-          && TEST_true(RSA_set0_key(key, n, e, d))
-          && TEST_true(RSA_set0_factors(key, p, q))
-
+          && TEST_true(RSA_set0_key(key, n, e, d));
+    if (!ret) {
+
+        BN_free(p);
+        BN_free(q);
+        BN_free(e);
+        BN_free(n);
+        BN_free(d);
+        goto end;
+    }
+    ret = TEST_true(RSA_set0_factors(key, p, q))
```

Comment:
If this call fails we leak `p` and `q`

Reply:
Should be fixed now. Not sure what I was thinking. Reordered some of the code so free of same pointer is not required in multiple places.